hash_practice/exercises.py

```diff
@@ -2,18 +2,49 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n log m)
+        Space Complexity: O(n)
     """
-    pass
+    if len(strings) < 1:
+        return []
+
+    anagrams = dict()
+
+    for string in strings:
+        key_string = "".join(sorted(string))
+
+        if key_string in anagrams:
+            anagrams[key_string].append(string)
+        else:
+            anagrams[key_string] = [string]
+
+    return [anagrams[group] for group in anagrams]
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: At least O(n) assuming that max() is an O(1) look up but I strongly suspect it isn't
+        Space Complexity: At least O(n + k)
+        (Not sure about the big O here becuase I'm not certain how max works in this case)
     """

```

Comment:
👍  Because you have a loop going k times and each time you find a maximum of the `n` elements.  I would say this is O(nk)